base/client.jl

```diff
@@ -125,14 +125,26 @@ function repl_cmd(cmd, out)
     nothing
 end
 
-display_error(er) = display_error(er, [])
-function display_error(er, bt)
-    print_with_color(Base.error_color(), STDERR, "ERROR: "; bold = true)
-    with_output_color(Base.error_color(), STDERR) do io
-        showerror(io, er, bt)
+function display_error(io::IO, er, bt)
+    if !isempty(bt)
+        st = stacktrace(bt)
+        if !isempty(st)
+            io = redirect(io, log_error_to, st[1])
+        end
+    end
+    print_with_color(Base.error_color(), io, "ERROR: "; bold = true)
+    # remove REPL-related frames from interactive printing
+    eval_ind = findlast(addr->Base.REPL.ip_matches_func(addr, :eval), bt)
+    if eval_ind != 0
+        bt = bt[1:eval_ind-1]
+    end
+    with_output_color(Base.error_color(), io) do io
+        showerror(IOContext(io, :limit => true), er, bt)
```

Comment:
my assumption was that whoever added `:limit => true` to `REPL.jl` simply forgot to add it to `client.jl` as well.  according to the `IOContext` docstring it simply truncates long containers with ellipses.

as an alternative, we could continue to keep maintaining two versions of `display_error`.  i was just trying to remove some repetition figuring it'd make maintenance easier.